src/lib/ens.ts

```diff
@@ -0,0 +1,40 @@
+// @ts-ignore TODO: add type declarations for ensjs module
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+import memoize from 'lodash/memoize'
+import { getWeb3Provider } from 'lib/web3-provider'
+
+let maybeEns: ENS | null
+
+const web3Provider = getWeb3Provider()
+
+const getEnsInstance = (): ENS => {
+  if (!maybeEns) {
+    // getEnsAddress takes a magic number as string, networkId. 1 stands for mainnet
+    maybeEns = new ENS({ provider: web3Provider, ensAddress: getEnsAddress('1') })
```

Comment:
we actually have this concept of mainnet being `1` in our caip2 implementation

https://github.com/shapeshift/lib/blob/main/packages/caip/src/caip2/caip2.ts#L12

it might be worth quickly familiarizing yourself with the CAIP2 standard (blockchain id) here https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md

afaik - we are the first to actually implement these. 

we can also get the `caip2` from the eth chain adapter

https://github.com/shapeshift/lib/blob/main/packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts#L56

it would be nice to pass this caip2 through and parse the network from it, such that when we support multiple networks we're ahead of things.

Reply:
Done! I'm also capturing the namespace here, in case we're abstracting this functionality away to an external module (which I definitely think would make sense, but not in this PR which is already large enough as is).